watchman/views.py

```diff
@@ -5,26 +5,29 @@
 from django.http import Http404
 
 from jsonview.decorators import json_view
-from watchman.utils import get_checks
-from watchman.settings import WATCHMAN_CHECKS
 from watchman.decorators import token_required
+from watchman.utils import get_checks
 
 
 @token_required
 @json_view
 def status(request):
     response = {}
-    available_checks = frozenset(WATCHMAN_CHECKS)
+
+    check_list = None
+    skip_list = None
+
     if len(request.GET) > 0:
         if 'check' in request.GET:
-            possible_filters = frozenset(request.GET.getlist('check'))
-            available_checks &= possible_filters
+            check_list = request.GET.getlist('check')
         if 'skip' in request.GET:
-            skipped_filters = frozenset(request.GET.getlist('skip'))
-            available_checks -= skipped_filters
-    for func in get_checks(paths_to_checks=available_checks):
-        if callable(func):
-            response.update(func(request))
+            skip_list = request.GET.getlist('skip')
+
+    for check in get_checks(check_list=check_list, skip_list=skip_list):
+        if callable(check):
```

Comment:
So far we're using this in only two places (here, and the management command) which is not overly burdensome.. if we add a third I'll look at refactoring it.